streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -460,6 +460,15 @@ private void close(final boolean clean) {
      * source topic partitions, or if it is enforced to be processable
      */
     public boolean isProcessable(final long wallClockTime) {
+        if (state() == State.CLOSED || state() == State.CLOSING) {
+            // a task is only closing / closed when 1) task manager is closing, 2) a rebalance is undergoing;
+            // in either case we can just log it and move on without notifying the thread since the consumer
+            // would soon be updated to not return any records for this task anymore.
+            log.info("Stream task {} is already in {} state, skip adding records to it.", id(), state());
```

Comment:
Is this what you meant?

```suggestion
            log.info("Stream task {} is already in {} state, skip processing it.", id(), state());
```